src/auth/ThreeStepOwned.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+/// @notice Three-step single owner authorization mixin.
+/// @author SolBase (https://github.com/Sol-DAO/solbase/blob/main/src/auth/ThreeStepOwned.sol)
+/// @author Modified from Solmate (https://github.com/transmissions11/solmate/blob/main/src/auth/Owned.sol)
+abstract contract ThreeStepOwned {
+    /*//////////////////////////////////////////////////////////////
+                                 EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    event OwnerUpdateInitiated(address indexed user, address indexed ownerCandidate);
+
+    event OwnerUpdated(address indexed user, address indexed newOwner);
+
+    /*//////////////////////////////////////////////////////////////
+                            OWNERSHIP STORAGE
+    //////////////////////////////////////////////////////////////*/
+
+    address public owner;
+    
+    address internal _ownerCandidate;
+    
+    bool internal _ownerCandidateConfirmed;
+
+    modifier onlyOwner() virtual {
+        require(msg.sender == owner, "UNAUTHORIZED");
+
+        _;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                               CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(address _owner) {
+        owner = _owner;
+
+        emit OwnerUpdated(address(0), _owner);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                             OWNERSHIP LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function setOwner(address newOwner) public virtual onlyOwner {
+        _ownerCandidate = newOwner;
+
+        emit OwnerUpdateInitiated(msg.sender, newOwner);
+    }
+
+    function confirmOwner() public virtual {
```

Comment:
I think there should be a deadline after which an owner cannot be confirmed anymore. Think about the following scenario:
- I call set `setOwner`
- New owner confirms
- The community does not want a new owner and the Board of Directors push me out of the company
- We discuss it with the community for 2 weeks but without a result
- I just think: fuck it - and confirm the ownership transfer since I still have access to the private key.

Time can change circumstances and I think such important transfers should have a deadline.

Reply:
I think the issue there is that the current owner is just an EOA owned by someone that shouldn't. Even having the deadline, you can renounce to ownership using the PK, and then you fuck the entire ownership leaving it as address(0).

Anyways, aside of the example, I hear you. But again, you need the deadline because you are kind of not trusting the current owner, and after the deadline the current owner can still proposing new owners or renouncing to the ownership. Not sure how much a deadline will help here.

Reply:
the scenario was more sort of a simple illustration; there are much more sophisticated ones - my main message is: time matters and we should be careful with theoretically forever-pending governance updates.